File: core/emergency-manager/app/src/main/java/project/grp3/emergency/core/database/entities/FireEntity.java
package project.grp3.emergency.core.database.entities;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import org.hibernate.annotations.Cascade;

import javax.persistence.*;
import java.util.Date;
import java.util.Objects;

@Entity
@Table(name = "Fire")
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class FireEntity
{

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private Date detectionDate;
    private Date endDate;
    @ManyToOne
    @Cascade(org.hibernate.annotations.CascadeType.ALL)
    @JoinColumn(name = "fire_type_id")
    private FireTypeEntity type;
    @OneToOne
    @Cascade(org.hibernate.annotations.CascadeType.ALL)
    @JoinColumn(name = "ressource_id")
    private ResourceEntity resource;
    @ManyToOne
    @JoinColumn(name = "sensor_id")
    private SensorEntity sensor;

    @Override
    public boolean equals(Object o)
    {
        if (this == o) return true;
        if (!(o instanceof FireEntity)) return false;
        FireEntity that = (FireEntity) o;
        return Objects.equals(id, that.id);
    }

    @Override
    public int hashCode()
    {
        return Objects.hash(id);
    }

    public Long getId()
    {
        return id;
    }

    public void setId(Long id)
    {
        this.id = id;
    }

    public Date getDetectionDate()
    {
        return detectionDate;
    }

    public void setDetectionDate(Date detectionDate)
    {
        this.detectionDate = detectionDate;
    }

    public Date getEndDate()
    {
        return endDate;
    }

    public void setEndDate(Date endDate)
    {
        this.endDate = endDate;
    }

    public FireTypeEntity getType()
    {
        return type;
    }

    public void setType(FireTypeEntity type)
    {
        this.type = type;
    }

    public ResourceEntity getResource()
    {
        return resource;
    }

    public void setResource(ResourceEntity resource)
    {
        this.resource = resource;
    }


    public SensorEntity getSensor()
    {
        return sensor;
    }

    public void setSensor(SensorEntity sensor)
    {
        this.sensor = sensor;
    }
}

File: core/mqtt/app/src/database/entities/FireEntity.ts
import {Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn} from "typeorm";
import {SensorEntity} from "./SensorEntity";
import {FireType} from "../../../../../truck-manager/app/src/core/apis/emergency/models";
import {FireTypeEntity} from "./FireTypeEntity";


export type SensorEntityWithId =  Omit<SensorEntity, "id">


@Entity({name: "fire"})
export class FireEntity {

    @PrimaryGeneratedColumn()
    id: number;

    @Column({name: "fire_id"})
    fireId: number

    @Column()
    intensity: number

    @ManyToOne(() => SensorEntity, sensor => sensor.sensorId)
    @JoinColumn({name: "sensor_id"})
    sensor: SensorEntity

    @Column("timestamp")
    date: Date


    @ManyToOne(() => FireTypeEntity, fireType => fireType.fireTypeId)
    type: FireType
}
